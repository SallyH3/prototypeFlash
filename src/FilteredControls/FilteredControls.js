import React from 'react';

const FilteredControls = (props) => {
  return (
<section className='left-box'>
        <article className='left-title'>
        Which prototype methods do you want to practice?
        </article>
        <p className='left-subtitle'>...click on a method below and your card will display on the right!</p>
        <article className='button-container'>
        <button id='Iteration methods' onClick={props.iteration} className='button iteration'>Iteration Methods</button>
        <button id='Mutator methods' onClick={props.mutator}className='button mutator'>Mutator Methods</button>
        <button id='Accessor methods' onClick={props.accessor} className='button accessor'>Accessor Methods</button>
        </article>
        <hr></hr>
      </section>
  )
}

export default FilteredControls;