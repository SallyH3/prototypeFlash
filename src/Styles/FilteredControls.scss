//container for left side of app
.left-box {
  background-color: $white;
  font-family: $primary;
  font-size: 1.3em;
  margin-top: 10em;
  padding: .8em;
  text-align: center;
  width: 50%;
}

//'which methods' title that displays on left side of app
.left-title {
  font-size: 1.3em;
  margin-top: 1em;
}

//subtitle directions for user to click button 
.left-subtitle {
  color: hsla(253, 50%, 53%, 1);
  font-size: .9em;
  font-style: italic;
  &:hover {
    color: hsla(283, 59%, 55%, 1);
    font-size: 1em;
  }
}

.button-container {
  display: flex;
  justify-content: space-between;
  margin: 2em;
}

//method buttons
.button {
  @include button-rules;
  background-color: $lavender;
  border: 1px solid $primaryBorderColor;
  cursor: pointer;
  border-radius: .5em;
  font-family: $primary;
  font-size: .8em;
  height: 3.4em;
  letter-spacing: .1em;
  margin: 1em;
  padding: .5em;
  width: 10em;
}

//creates line under method buttons on left of page
hr {
  width: 80%;
}

//gets rid of outline after button click
.button:focus {outline:0}

//text on left side of app - bottom left 'study' area
.need-to-study-text {
  font-size: 1.2em;
  margin-top: 2em;
}
//local storage number area
.study-list-text {
  text-decoration: underline;
  margin-right: 3em;
}

@media screen and (max-width: 1004px) {
  .left-box {
    width: 50%;
  }

  hr {
    width: 75%;
  }
  
  .button-container {
    margin-left: 0px;
    margin-right: 0px;
    width: 50%;
  }
}

@media screen and (max-width: 793px) {
  .left-box,
  .button-container {
    display: flex;
    flex-direction: column;
    width: 100%;
  }

  .button {
    width: 75%;
  }

  .left-title,
  .need-to-study-text,
  .study-list-text {
    font-size: .9em;
  }
  
  .left-subtitle {
    font-size: .7em;
    &:hover {
      font-size: .8em;
    }
  }

  .button-container {
    width: 100%;
    margin: .8em;
  }
}

@media screen and (max-width: 575px) {
  .button {
    margin: 1em;
    width: 75%;
  }
}