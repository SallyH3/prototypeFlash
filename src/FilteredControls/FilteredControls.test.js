import React from 'react';
import FilteredControls from './FilteredControls';
import { shallow } from 'enzyme';

describe ('FilteredControls', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <FilteredControls
      />
    )
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});