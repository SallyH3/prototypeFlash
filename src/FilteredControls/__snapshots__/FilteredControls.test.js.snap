// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilteredControls should match snapshot 1`] = `
<section
  className="left-box"
>
  <article
    className="left-title"
  >
    Which prototype methods do you want to practice?
  </article>
  <p
    className="left-subtitle"
  >
    ...click on a method below and your card will display!
  </p>
  <article
    className="button-container"
  >
    <button
      className="button iteration "
      id="Iteration methods"
      name="iterator"
    >
      Iteration Methods
    </button>
    <button
      className="button mutator "
      id="Mutator methods"
      name="mutator"
    >
      Mutator Methods
    </button>
    <button
      className="button accessor "
      id="Accessor methods"
      name="accessor"
    >
      Accessor Methods
    </button>
  </article>
  <hr />
</section>
`;
